public/components/event_analytics/explorer/sidebar/sidebar.tsx

```diff
@@ -117,8 +129,35 @@ export const Sidebar = (props: ISidebarProps) => {
     [explorerFields, tabId]
   );
 
-  const onDragEnd = ({}) => {
-    console.log('source, destination');
+  // this useEffect will set allFields when the page is rendered
+  useEffect(() => {
+    if (
+      explorerFields.availableFields.length > 0 &&
+      explorerFields.selectedFields.length === 0 &&
+      allFields.length === 0
+    ) {
+      setAllFields(explorerFields.availableFields);
+    }
+  }, [explorerFields.availableFields]);
+
+  const onDragEnd = ({
```

Comment:
did we include drag and drop changes?

Reply:
It was mostly implemented already by Sumukh, all I did was add the logic for a field being added/removed. Your choice as to whether we implement or not, I can remove it